viewing_party/commented_party.py

```diff
@@ -0,0 +1,184 @@
+# # Wave
+
+# 1.create movie
+# if parameters are truthy, retrurn a dict
+# dict should have three key-value pairs keys: tittle, genre, and rating
+def create_movie(title, genre, rating):
+    movie = {}
+    if title and genre and rating:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+    else: 
+        movie = None
+    return movie
+
+# 2. user_data is a dict key-"watched" and val-list of dict(movie)
+# (empty list means user ha no movies in watched list)
+# add movie to watched list inside of user_data and return user data
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+# 3. add movie(dict) to user_data indexed at "watchlist"
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+# 4. if title(str) is a movie in "watchlist", 
+# remove movie(dict) from "watchlist"
+# and add to "watched"
+# return user_data
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
+
+
+# # Wave 2
+
+# 1. calculate avg rating for movies in "watched" within user_data(dict)
+def get_watched_avg_rating(user_data):
+    watched_ratings = []
+    for movie in user_data["watched"]:
+        watched_ratings.append(movie["rating"])
+    number_watched = len(watched_ratings)
+    if number_watched:
+        watched_avg_rating = sum(watched_ratings)/number_watched
+    else:
+        watched_avg_rating = 0.0
+    return watched_avg_rating
+
+# 2. for movie in user_data["watched"] add genre as key to genres dict
+#    value at genres["genre"] is +1 every time genre appears
+def get_most_watched_genre(user_data):
+    watched_genres = {}
+    # if we've watched any movies (else return None)
+    if user_data["watched"]:
+        # populate watched_genres freq map
+        for movie in user_data["watched"]:
+                genre = movie["genre"]
+                if genre in watched_genres:
+                    watched_genres[genre] += 1
+                else:
+                    watched_genres[genre] = 1
+        # find max frequency and paired genre
+        max_watched_genre_freq = max(watched_genres.values())
+        for genre, freq in watched_genres.items():
+            if freq == max_watched_genre_freq:
+                most_watched_genre = genre
+        return most_watched_genre
```

Comment:
You can clean this up a little bit with an early return (personally I find extra indentation makes functions a little harder to read):

```suggestion
    if not user_data["watched"]:
        return None
    # populate watched_genres freq map
    for movie in user_data["watched"]:
            genre = movie["genre"]
            if genre in watched_genres:
                watched_genres[genre] += 1
            else:
                watched_genres[genre] = 1
    # find max frequency and paired genre
    max_watched_genre_freq = max(watched_genres.values())
    for genre, freq in watched_genres.items():
        if freq == max_watched_genre_freq:
            most_watched_genre = genre
    return most_watched_genre
```